### PR TITLE
feat: add HomePage

### PR DESCRIPTION
**Pull request by Shailseen**
**Features:**
Aside bar with `Icons Only` 
Aside bar With `Icons and Text`
Responsive For Mobiles also
image
image

### CODE REVIEW

**Review by surbhi84 — APPROVED** [review]
The code looks neat.
Good use of react components.
Didn't find any breaking changes, looks good to merge.
suggestions- make use of absolute paths and try to put more than one commit for this much code.

File: src/components/Asides/Aside.jsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { useAside } from "../../context/aside-context";
```
Comment (arpit01923) [inline src/components/Asides/Aside.jsx:1]: use absolute path to remove ../../

File: src/components/Asides/IconAside/IconAside.jsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import classNames from "classnames"
+import iconStyle from "./IconAside.module.css"
+import styles from "../OpenAside/OpenAside.module.css"
+import HistoryRoundedIcon from "@mui/icons-material/HistoryRounded";
+import ThumbUpOutlinedIcon from '@mui/icons-material/ThumbUpOutlined';
+import VideoLibraryOutlinedIcon from '@mui/icons-material/VideoLibraryOutlined';
+import HomeOutlinedIcon from '@mui/icons-material/HomeOutlined';
+import WatchLaterOutlinedIcon from '@mui/icons-material/WatchLaterOutlined';
+import {useAside} from "../../../context/aside-context"
```
Comment (arpit01923) [inline src/components/Asides/IconAside/IconAside.jsx:9]: same use absolute path

File: src/components/Asides/MobileAside/MobileAside.jsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import styles from "./MobileAside.module.css";
+import HistoryRoundedIcon from "@mui/icons-material/HistoryRounded";
```
Comment (arpit01923) [inline src/components/Asides/MobileAside/MobileAside.jsx:2]: maybe barrel imports is good choice

File: src/components/Asides/OpenAside/OpenAside.jsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import classNames from "classnames";
+import styles from "./OpenAside.module.css";
+import HistoryRoundedIcon from "@mui/icons-material/HistoryRounded";
+import ThumbUpOutlinedIcon from "@mui/icons-material/ThumbUpOutlined";
+import VideoLibraryOutlinedIcon from "@mui/icons-material/VideoLibraryOutlined";
+import HomeOutlinedIcon from "@mui/icons-material/HomeOutlined";
+import WatchLaterOutlinedIcon from "@mui/icons-material/WatchLaterOutlined";
+import { useAside } from "../../../context/aside-context";
```
Comment (arpit01923) [inline src/components/Asides/OpenAside/OpenAside.jsx:8]: same use absolute paths

**Review by arpit01923 — APPROVED** [review]
- clean UI
- good folder structure
- work on suggestion

File: src/components/Asides/OpenAside/OpenAside.jsx (hunk 4 above)
Comment (Shailseen) [reply src/components/Asides/OpenAside/OpenAside.jsx:8]: i will do these on later.

**Pull request merged**